### ISSUE TITLE
Assigning sub-class to parent class with generic typed function property does not work

### ISSUE THREAD

**Issue by skrtheboss** [microsoft/TypeScript#35748]
**TypeScript Version:**  3.7.2

**Search Terms:**
class, extend, function, generic

**Code**

```ts
interface ITest {
    prop1: string;
}

class Test<T extends ITest> {
    protected test: T | null = null; 
    protected test1 = (value: T) => undefined;
    protected test2(value: T): void { }
}

interface ITestExt extends ITest {
    prop2: boolean;
}

class TestExt<T extends ITestExt> extends Test<T> {
}

const testExt = new TestExt<ITestExt>();

function test<T extends Test<ITest>>(api: T): void {
}

test(testExt);

```

**Expected behavior:**
Assigning `TestExt` to `Test` should not result in a type error.
If the `test1` property is removed from `Test` then no error is shown.

**Actual behavior:**
Assigning `TestExt` to `Test` results in an error.

**Playground Link:** 
[TypescriptPlayGround](http://www.typescriptlang.org/play/?target=6&module=1&ssl=1&ssc=1&pln=24&pc=1#code/JYOwLgpgTgZghgYwgAgJIBUIGczIN4BQyxyADlAPakCMAXMjlKAOYDcBAvgQQgDZxYsyTDgA86ZBAAekEABMhGbGAB8+IiXIVICSHOSQc9CQB9kIAK69eyALzmrvVsg3EtOvQeXU7yABQAbnC8FhDGAJR2ahbyEDCgEHLsJGSUHoleOABMgcGhEfQBFMD6eMhcXASgkLCIKEo4AKIykjIQ8ooiuIQpWqRZ9ABGFBS8EHAg7JV8AkJdzWDirbIKaPMyatIrc8riaoTTFCA4mWALviAQAO7CyguiDWcbfuHsBDAxusBHp0tb7asug8uioVH44KRgAVkEUSupKoYwH5EQtXgQgA)

**Related Issues:**
-

**Comment by ilogico** [comment]
I believe this is working as intended.
Functions declared as properties are sound, while functions declared as methods are bivariant.
Bear in mind that `(v: ItestExt) => void` is not assignable to `(v: Itest) => void`, this is enforced by the `strictFunctionTypes` flag (which doesn't affect functions declared as methods, that's why `test2` typechecks.
